docs/guide/mutating_datasets.rst

```diff
@@ -324,3 +326,46 @@ When garbage collection is called, the files are removed.
     files_before.difference(store.keys())  # Show files removed
 
 .. _storefact: https://github.com/blue-yonder/storefact
+
+
+Mutating indexed datasets
+-------------------------
+
+If the to-be-updated dataset was created with an index, every update on this dataset will update the index automatically and ensure the dataset will never end up in an inconsistent state. This even holds true in case the update function does not specify any or only partially the indices. Consider the following example
```

Comment:
I think this can be made much clearer so the wording also applies much cleaner to:

 * cases in which indices were created after dataset creation
 * cases with more than one index

```suggestion
The mutating operation will update all indices that currently exist for the dataset. This even holds true in case the update function does not specify any or only partially the indices. Consider the following example
```